### PR TITLE
Intent slot tutorial

### PR DESCRIPTION
**Pull request by vladgets**
Added evaluation part and few corrections.
Also changed the name of the notebook to JointIntentSlotClassification

### CODE REVIEW

File: nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def get_features(
 
     logging.info(f'{too_long_count} are longer than {max_seq_length}')
 
+    ''' # May be useful for debugging
```
Comment (ekmb) [inline nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py:115]: You can use logging.debug for this purpose and remove the comment

File: tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb [hunk 2]
```diff
@@ -76,7 +76,7 @@
...
+    "Nemo provides **import_dataset.py** converter for few reference datasets (Assistant/Atis/Snips) which converts them to the Nemo data format for the Intent and Slot classification model. If you have your own annotated dataset in a different format, you will need to write a data converter. Possible recommended format for your own annotation, is to have one text file per all examples of one intent. With one line per query in a form like: `did i set an alarm to [alarm_type : wake up] in the [timeofday : morning]`, using brackets to define slot names. This is very similar to the assistant format from this example and you can use its converter to Nemo format with small changes. \n",
```
Comment (ekmb) [inline tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb:79]: Nemo -> NeMo

File: tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb [hunk 3]
```diff
@@ -482,23 +474,53 @@
...
    "source": [
-    "## Training Script\n",
-    "\n",
-    "If you have NeMo installed locally, you can also train the model with the example script: `examples/nlp/intent_slot_classification/intent_slot_classification.py.`\n",
-    "\n",
-    "To run a training script, use:\n",
-    "\n",
-    "`cd examples/nlp/intent_slot_classification`\n",
+    "## Evaluation\n",
+    "To see how the model performs, we can run evaluate and test the performance of the trained model on a data file. Here we would load the best checkpoint (the one with the lowest validation loss) and create a model (eval_model) from the checkpoint. We will use the same trainer for testing."
```
Comment (ekmb) [inline tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb:478]: run evaluate -> evaluate / run evaluation

**Review by ekmb — CHANGES_REQUESTED** [review]

File: tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb (hunk 2 above)
Comment (VahidooX) [inline tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb:79]: NeMo, not Nemo.

Comment (VahidooX) [inline tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb:79]: Do we support this format directly in nemo model?

**Review by VahidooX — CHANGES_REQUESTED** [review]

File: nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py (hunk 1 above)
Comment (vladgets) [reply nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py:115]: Corrected, thanks

File: tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb (hunk 2 above)
Comment (vladgets) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb:79]: Corrected

File: tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb (hunk 3 above)
Comment (vladgets) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb:478]: Corrected

File: tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb (hunk 2 above)
Comment (vladgets) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb:79]: Corrected in several places, thank you.

Comment (vladgets) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb:79]: Which one from 3 you are asking? We do have converter mode for each 3 of them to Nemo data format.

Comment (VahidooX) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb]: So here we are suggesting users to create their data in Assitant format and them convert it to NeMo's format?

Comment (VahidooX) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb]: If yes, lets make it more clear here. And are you sure if they gather their data, they can easily use the same convertor? I mean that one is written for that specific dataset and folder structure?

Comment (vladgets) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb]: I wanted to do that, but there is a small issue. This Assistant data format also it is good, it keeps multiple columns in their original json files. Like Query before manual correction, after manual correction, with annotation, etc. So in my converter, I am using a single columns (separated by ";") for each line of their json files. And although I do like their annotation format it is convoluted with other non relevant data in the same file. So, I did not know how to explain it better to user?

Comment (VahidooX) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb]: This is something that we should work on, but it looks to me it is not ready. So how about dropping this suggestion here for now and come up with a good plan later?

Comment (VahidooX) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb]: This beta version is a good opportunity to reevaluate and redesign the nemo format if you feel like it is not good enough.

Comment (vladgets) [reply tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb]: I sent you my more detailed response in Slack

**Review by VahidooX — APPROVED** [review]

**Pull request merged**